### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'map')

### ISSUE THREAD

**Issue by krsaedan** [ssut/nestjs-sqs#72]
Hello, currently, I have the following error when trying to send messages: "Cannot read properties of undefined (reading 'map')" when executing "this.sqsService.send".

The configuration is as follows:

"@ssut/nestjs-sqs": "^1.2.0",
"aws-sdk": "^2.1488.0",

the error is intermittent

**Comment by Noezor** [comment]
Getting the same issue. Are you sure it's from the library? The intermittence of the issue is weird.

**Comment by gabriel-skinny** [comment]
Happining the same here. Started intermittent, but now every time i call this method it returns the same error. I think it is probally the version that im using (1.0.1), because what is causing the error is in the line 48 of producer.js, and this parte of the code only  treats the return of the sqs message. I think they changed the return and the old versions are not working.

**Comment by gabriel-skinny** [comment]
Yep, its seems that is it. The aws-sdk changed the return of the sendMessageBatch function, and does not return the object of failure. In the newest version of sqs-producer, they changed the treatment, but in the old ones are not working. One solution is to downgrade aws-sdk to the version 2.1490, or you can update the sqs-producer to 2.0.2.

There is already an issue open on the aws-sdk repository: https://github.com/aws/aws-sdk-js/issues/4534.

**Comment by Noezor** [comment]
I think we can close as not related to this library ?

**Comment by sildar** [comment]
I believe the issue is with the version of sqs-producer that is used, as the last version of that library [does check for undefined](https://github.com/bbc/sqs-producer/blob/d549400b657bb252dc8ddd2047c37f5df207a87e/src/producer.ts#L89).
